lib/test-runner-process.coffee

```diff
@@ -0,0 +1,31 @@
+Parser = require 'tap-parser'
+TerminalCommandExecutor = require './terminal-command-executor'
+
+module.exports =
+  class TestRunnerProcess
+    constructor: (executor = new TerminalCommandExecutor) ->
+      @terminalCommandExecutor = executor
+      @terminalCommandExecutor.onDataReceived (data) => @addAvaOutput(data)
+      @terminalCommandExecutor.onDataFinished => @endAvaOutput()
+
+    run: (filePath, assertCallback, completeCallback) ->
+      @parser = @getParser()
+      @parser.on('assert', assertCallback)
+      @parser.on('complete', completeCallback)
+
+      #TODO: Fix parsing of folders and files
+      folder = filePath.substring(0, filePath.lastIndexOf("/") + 1);
+      file = filePath.split("/").pop()
+
+      command = "ava #{file} --tap"
```

Comment:
@sindresorhus, sorry if I'm missing something here.

When trying to use the programmatic API providing an absolute path for running a test I'm getting 'No tests found in ../x/y.js, make sure to import "ava" at the top of your test file'. Same result if I use the CLI in a folder different from the one that hosts the file, as: 

- ava /Users/Me/Folder/test.js --verbose

I have been able to trace the code till test-worker.js but being honest, I don't know why it's failing. Any idea?. Maybe I'm doing something wrong...

Thank you! :)